dockerfile fixes after hadolint scan

tkatila opened the pull request:
Signed-off-by: Tuomas Katila <tuomas.katila@intel.com>

tkatila commented:
If/when the fixes are deemed ok, I'll "make dockerfiles" for the full effect.

mythi commented on build/docker/lib/toybox_build.docker:
```diff
@@ -3,7 +3,8 @@ ARG TOYBOX_SHA256="2bed6bb9edd5a249023103cf0402a835b0e53d10304a263f6f1e77a8aa49a
 ARG ROOT=/install_root
 
-RUN apt update && apt -y install musl musl-tools musl-dev
+RUN apt-get update && apt-get --no-install-recommends -y install musl=1.2.2-1 musl-tools=1.2.2-1 musl-dev=1.2.2-1
```
I'd prefer we don't lock the versions even if the tool complains:
```suggestion
RUN apt-get update && apt-get --no-install-recommends -y install musl musl-tools musl-dev
```

bart0sh replied:
I can still see versions for all packages? Are you going to apply @mythi's suggestion?

bart0sh commented:
@tkatila Dockerfile* and Dockerfile.in* are out of sync it seems. Here is what CI check reports:
 ```
Please commit result of "make dockerfiles" to make them in synch with the .in -files.
Error: Process completed with exit code 2.
```

bart0sh commented on build/docker/lib/default_build.docker:
```diff
@@ -1,8 +1,10 @@
+WORKDIR ${DIR}/cmd/${CMD}
+RUN GO111MODULE=${GO111MODULE} CGO_ENABLED=0 go install "${BUILDFLAGS}"
+WORKDIR ${DIR}
+RUN install -D /go/bin/${CMD} /install_root${EP}
```
This would increase number of layers from one to 4. What's the benefit?

tkatila replied:
Removal of a few hadolint warnings. But I don't think it's cost beneficial to add layers to remove a warning in this case. Thanks for noticing it!

tkatila commented:
> @tkatila Dockerfile* and Dockerfile.in* are out of sync it seems. Here is what CI check reports:
> 
> ```
> Please commit result of "make dockerfiles" to make them in synch with the .in -files.
> Error: Process completed with exit code 2.
> ```

Yes, I know. I wanted to propose these changes via the .in files first. And once the set of changes are agreed upon, I'd add the full .Dockerfile changeset.

mythi commented on build/docker/intel-deviceplugin-operator.Dockerfile:
```diff
@@ -25,7 +25,7 @@ ARG CMD=operator
 ## 2) and the default FINAL_BASE is primarily used to build Redhat Certified Openshift Operator container images that must be UBI based.
 ## The RedHat build tool does not allow additional image build parameters.
-ARG FINAL_BASE=registry.access.redhat.com/ubi8-micro
+ARG FINAL_BASE=registry.access.redhat.com/ubi8-micro:8.6
```
@chaitanya1731 is this selection OK for you?

chaitanya1731 replied:
Looks good to me.

bart0sh commented on build/docker/intel-fpga-initcontainer.Dockerfile:
```diff
@@ -75,7 +73,8 @@ RUN echo "{\n\
 ARG ROOT=/install_root
-RUN apt update && apt -y install musl musl-tools musl-dev
+RUN apt-get update && apt-get --no-install-recommends -y install musl musl-tools musl-dev
+SHELL ["/bin/bash", "-o", "pipefail", "-c"]
```
Just curious, what this is for?

tkatila replied:
https://github.com/hadolint/hadolint/wiki/DL4006

mythi reviewed (approved):
thanks!

hj-johannes-lee commented:
Sorry, accidently I canceled your pr's ci test run..! :(

tkatila commented:
> Sorry, accidently I canceled your pr's ci test run..! :(

It's probably fine. The previous version was tested and my last commit shouldn't change anything.

bart0sh commented on build/docker/lib/final_base.docker:
```diff
@@ -8,5 +8,5 @@
 ## 2) and the default FINAL_BASE is primarily used to build Redhat Certified Openshift Operator container images that must be UBI based.
 ## The RedHat build tool does not allow additional image build parameters.
-ARG FINAL_BASE=registry.access.redhat.com/ubi8-micro
+ARG FINAL_BASE=registry.access.redhat.com/ubi8-micro:8.6
```
Can you point me out to the explanation about this one?

bart0sh replied:
nvm, was looking at the old change.

bart0sh reviewed (dismissed):
/lgtm

bart0sh commented:
@tkatila can you squash your commits? I think it would be more readable when the same kind of changes are not spread on multiple commits.

tkatila commented:
> @tkatila can you squash your commits? I think it would be more readable when the same kind of changes are not spread on multiple commits.

Yes! Squashed and rebased.
